### ISSUE TITLE
pmac32_defconfig - CONFIG_HIGHMEM seems to have no effect in >=5.11

### ISSUE THREAD

**Issue by kth5** [linuxppc/issues#399]
I have a 17" PowerBook G4 (PowerBook5,5) with 1.5GB RAM (1GB + 512MB) but the machine sees only ~736MB.

```
[kth5@pbg4 ~]$ zgrep "CONFIG_HIGHMEM" /proc/config.gz 
CONFIG_HIGHMEM=y
[kth5@pbg4 ~]$ free -m
               total        used        free      shared  buff/cache   available
Mem:             735          41         570           0         122         680
Swap:           4566           0        4566
```

I do have **CONFIG_HIGHMEM**, **CONFIG_MMU** and **CONFIG_BOUNCE** enabled to no effect in the following **.config**:

https://github.com/kth5/archpower/blob/master/linux/config.ppc32

Assuming it's a problem in my config, I rebuilt a kernel with **pmac32_defconfig** which also leads to the same while the machine **/proc/cpuinfo** indicates the amount of RAM correctly:

```
processor       : 0
cpu             : 7447A, altivec supported
clock           : 749.999000MHz
revision        : 1.1 (pvr 8003 0101)
bogomips        : 36.86

total bogomips  : 36.86
timebase        : 18432000
platform        : PowerMac
model           : PowerBook5,5
machine         : PowerBook5,5
motherboard     : PowerBook5,5 MacRISC3 Power Macintosh 
detected as     : 287 (PowerBook G4 17")
pmac flags      : 0000001a
L2 cache        : 512K unified
pmac-generation : NewWorld
Memory          : 1536 MB
```

The last known version that works for me with HIGHMEM properly is **5.10.108**.

Since I can't find a working **.config** to build my own, I also tried the Debian powerpc{,-smp} kernel which show the same effect.

**Comment by chleroy** [comment]
Can you try with the two patches at https://patchwork.ozlabs.org/project/linuxppc-dev/list/?series=286464&state=*

**Comment by kth5** [comment]
Applied both patches to my build of 5.17 and seems to now work properly and I can see the total RAM in my machine. Will run a few tests to ensure stability with higher memory usage but otherwise I'd say, that was it.

Thank you @chleroy! Appreciate the quick help!

**Comment by kth5** [comment]
After about an hour with `stress` everything staid stable. These two patches fixed my issue!

I wonder though, how nobody noticed this before releasing kernels for so long.

**Comment by chleroy** [comment]
Fixed by https://github.com/linuxppc/linux/commit/1ff5c8e8c835e8a81c0868e3050c76563dd56a2c

**Comment by mpe** [comment]
Fixed in v5.17.3 by gregkh/linux@9dad6b7dad3e.

Fixed in v5.16.20 by gregkh/linux@78c47402a89b.

**Comment by mpe** [comment]
See also: https://bugzilla.kernel.org/show_bug.cgi?id=215781
